src/loaders/express.js

```diff
@@ -55,6 +56,34 @@ module.exports = async (app) => {
     AWSXRay.getSegment().addMetadata('podName', config.podName);
     next();
   });
+  const authenticationEnabled = false;
+  if (authenticationEnabled) {
+    app.use(async (req, res, next) => {
+      let workRequest = [];
+      try {
+        // this try is so that the api doesn't crash if there's nothing in the body
+        workRequest = JSON.parse(JSON.parse(req.body).Message).request || false;
+      // eslint-disable-next-line no-empty
+      } catch (err) {}
+      const bearerHeader = req.headers.authorization
+      || workRequest.extraHeaders.Authorization;
+      const url = req.url.split('/');
+
+      const experimentId = url[url.indexOf('experiments') + 1] || workRequest.experimentId;
```

Comment:
I am thinking here that the part `|| workRequest.experimentId` might never get executed, because the part `url[url.indexOf('experiments') + 1]` will return something, and that for us there is no way to know if that something is the experiment id or not. For particular example, if the url was `https://scp.biomage.net/experiments/data-exploration` then the code here will "think" that `data-exploration` is the experiment id. Now, obviously right now we don't format this like that, but my worry is that if in the future we changed the way we compose the URL, this piece of code will silently fail. Do you know what I mean?